executor/builder.go

```diff
@@ -740,14 +739,6 @@ func (b *executorBuilder) buildShow(v *plannercore.PhysicalShow) Executor {
 		GlobalScope:  v.GlobalScope,
 		Extended:     v.Extended,
 	}
-	if e.Tp == ast.ShowGrants && e.User == nil {
```

Comment:
Why not put it into `fetchShowGrants`?

Reply:
I did move it to `fetchShowGrants`. Did you mean why did I move it?

I wanted to move the logic to one place because the handling of `SHOW GRANTS` versus `SHOW GRANTS FOR 'x'` is confusing. They are not really the same statement at all. The comment here is also incorrect, it looks like `show grants for current_user()` is the same as `show grants`.